app/models/search_result.rb

```diff
@@ -2,8 +2,11 @@
 
 class SearchResult < ApplicationRecord
   enum search_engine: { google: 'google', bing: 'bing' }
+  enum status: { in_progress: 'in_progress', completed: 'completed', failed: 'failed' }
 
   validates :keyword, presence: true
   validates :search_engine, presence: true
```

Comment:
It is best to combine presence validators into one line when having 3 or more attributes using the same validation type:

```suggestion
  validates :keyword, :search_engine, :status, presence: true
```